### PR TITLE
Makes the SBR lease name configurable

### PR DESCRIPTION
**Pull request by ignasi35**
Refs https://github.com/akka/akka-platform-guide/pull/583

### CODE REVIEW

File: akka-cluster/src/main/resources/reference.conf [hunk 1]
```diff
@@ -487,6 +487,12 @@ akka.cluster.split-brain-resolver.keep-oldest {
 akka.cluster.split-brain-resolver.lease-majority {
   lease-implementation = ""
 
+  # The identifier of the lease.
+  #  lease-name = "<service-name>-akka-sbr"
+  # When not set, it default to
+  #  lease-name = "<actor-system-name>-akka-sbr"
+  #lease-name = ""
```
Comment (patriknw) [inline akka-cluster/src/main/resources/reference.conf:494]: In Akka we typically use `""` as the undefined value. Better to have all properties in reference.conf and avoid `hasPath` stuff (some tooling depend on that such as config checker, and maybe also IntelliJ)

**Review by patriknw — COMMENTED** [review]
looking good

**Review by chbatey — APPROVED** [review]
LGTM

ActorSystem can probably be an invalid CR name in Kubernetes  so good to be able to override it

File: akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverSettings.scala [hunk 2]
```diff
@@ -96,7 +96,15 @@ import akka.util.Helpers.Requiring
     val acquireLeaseDelayForMinority =
       FiniteDuration(c.getDuration("acquire-lease-delay-for-minority").toMillis, TimeUnit.MILLISECONDS)
 
-    LeaseMajoritySettings(leaseImplementation, acquireLeaseDelayForMinority, role(c))
+    val leaseName = if (c.hasPath("lease-name")) {
```
Comment (patriknw) [inline akka-cluster/src/main/scala/akka/cluster/sbr/SplitBrainResolverSettings.scala:99]: hasPath not needed now

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by johanandren** [comment]
Needs formatting of the multi-jvm sources to be ready for merge.

**Pull request merged**